compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -492,6 +517,12 @@ private void findTypeParamInUnion(Location loc, BType expType, BUnionType actual
             if (type.tag == TypeTags.MAP) {
                 members.add(((BMapType) type).constraint);
             }
+            if (TypeTags.isXMLTypeTag(type.tag)) {
+                if (type.tag == TypeTags.XML) {
+                    members.add(((BXMLType) type).constraint);
+                }
+                members.add(type);
```

Comment:
Do we need to add the type also when `type.tag == TypeTags.XML` is true?

Reply:
Tracked in issue: #27851